### PR TITLE
dts: nxp: Convert ADC, I2C, SPI drivers to new DT_INST macros

### PR DESCRIPTION
**Pull request by galak**
Convert the following drivers:
```
 adc/adc_mcux_adc12.c
 adc/adc_mcux_adc16.c 
 i2c/i2c_mcux.c
 i2c/i2c_mcux_lpi2c.c
 spi/spi_mcux_dspi.c
 spi/spi_mcux_flexcomm.c
 spi/spi_mcux_lpspi.c
```

### CODE REVIEW

File: drivers/adc/adc_mcux_adc12.c [hunk 1]
```diff
@@ -254,134 +254,49 @@ static const struct adc_driver_api mcux_adc12_driver_api = {
...
-		    &mcux_adc12_init, &mcux_adc12_data_2, &mcux_adc12_config_2,
-		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
-		    &mcux_adc12_driver_api);
-
-static void mcux_adc12_config_func_2(struct device *dev)
-{
-	IRQ_CONNECT(DT_INST_IRQN(2),
-		    DT_INST_IRQ(2, priority), mcux_adc12_isr,
-		    DEVICE_GET(mcux_adc12_2), 0);
+#define ADC12_REF_SCR(n)						\
```
Comment (henrikbrixandersen) [inline drivers/adc/adc_mcux_adc12.c:257]: Did you mean to call the macro `ADC12_REF_SCR` or was it a typo? `ADC12_REF_SRC`?

**Review by henrikbrixandersen — APPROVED** [review]

File: drivers/adc/adc_mcux_adc12.c (hunk 1 above)
Comment (galak) [reply drivers/adc/adc_mcux_adc12.c:257]: typo, will fix.

**Comment by galak** [comment]
Pulled UART drivers out to hopefully get this to pass CI.

File: drivers/adc/adc_mcux_adc12.c (hunk 1 above)
Comment (MaureenHelm) [reply drivers/adc/adc_mcux_adc12.c:257]: this still needs to be done

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
just one nit from @henrikbrixandersen

File: drivers/adc/adc_mcux_adc12.c (hunk 1 above)
Comment (galak) [reply drivers/adc/adc_mcux_adc12.c]: I swore I did that, oh well. done now :)

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**